org.eclipse.acute/src/org/eclipse/acute/dotnetnew/DotnetNewWizard.java

```diff
@@ -38,6 +47,27 @@
 	@Override
 	public void init(IWorkbench workbench, IStructuredSelection selection) {
 		wizardPage = new DotnetNewWizardPage();
+
+		Iterator<Object> selectionIterator = selection.iterator();
+		Set<IWorkingSet> workingSets = new HashSet<>();
+		File selectedResource = null;
+
+		while (selectionIterator.hasNext()) {
+			Object element = selectionIterator.next();
+			File asFile = toFile(element);
+
+			if (asFile != null && selectedResource == null) {
+				selectedResource = asFile;
+				workingSets.addAll(getWorkingSets(element));
```

Comment:
If the user selects one folder and one working set, they most likely want the folder to be in the selected working set rather than to have the working set inferred from the element. So I would move that block to line 68 as
```java
if (workingSets.isEmpty() && selectedResource != null) {
  workingSets.addAll(getWorkingSets(selectedResource)
}
```